sklearn/base.py

```diff
@@ -438,8 +439,14 @@ def _validate_data(self, X, y=None, reset=True,
         return out
 
     def _repr_html_(self):
-        """HTML representation of estimator"""
-        return estimator_html_repr(self)
+        """HTML representation of estimator.
+
+        This is defined to display the HTML representation in sphinx-gallery.
+        Jupyer kernels will use `_repr_mimebundle_`.
+        """
+        if get_config()["display"] == 'diagram':
+            return estimator_html_repr(self)
+        return f"<code>{html.escape(repr(self))}</code>"
```

Comment:
Can't we just raise an AttibuteError here, shouldn't it be handled by sphinx gallery etc?